indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```diff
@@ -83,6 +92,47 @@ protected void map(
     }
   }
 
-  abstract protected void innerMap(InputRow inputRow, Text text, Context context)
+  public final static InputRow parseInputRow(Writable value, InputRowParser parser)
+  {
+    if(value instanceof Text) {
+      //Note: this hack is required because StringInputRowParser is a
+      //InputRowParser<ByteBuffer> and not InputRowParser<String>
+      //parse() method on it expects a ByteBuffer and not String
+      ByteBuffer bb = ByteBuffer.wrap(value.toString().getBytes());
+      return parser.parse(bb);
```

Comment:
We're assuming here that StringInputRowParser is somehow intrinsically tied to having a Text input. Is that assumption enforced anywhere?

If not can you please add a sanity check here to make sure future developers don't miss that design constraint?